ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java

```diff
@@ -559,6 +568,43 @@ public void applyIndexedAttestations(final List<ValidateableAttestation> attesta
         .reportExceptions();
   }
 
+  private void applyAttesterSlashingsFromBlock(final SignedBeaconBlock signedBeaconBlock) {
+    signedBeaconBlock
+        .getBeaconBlock()
+        .ifPresent(
+            beaconBlock ->
+                beaconBlock
+                    .getBody()
+                    .getAttesterSlashings()
+                    .forEach(
+                        attesterSlashing ->
+                            onAttesterSlashing(
+                                attesterSlashing, InternalValidationResult.ACCEPT, true)));
```

Comment:
We can just use signedBeaconBlock.getMessage() here and avoid the optional.

```suggestion
    signedBeaconBlock
        .getMessage()
        .getBody()
        .getAttesterSlashings()
        .forEach(
            attesterSlashing ->
                onAttesterSlashing(attesterSlashing, InternalValidationResult.ACCEPT, true));
```

Reply:
Actually, given we already create a vote update transaction in `applyVotesFromBlock` we should combine the two so that all the updates are done in a single transaction rather than two.  We can just extract the code from `onAttesterSlashing` that actually processes the slashing into another method so it can be reused.